src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/ContainerBuilderExtensions.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using Autofac;
+using Microsoft.Extensions.Logging;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    public static class ContainerBuilderExtensions
+    {
+        public static void RegisterLogging(this ContainerBuilder builder, ILoggerFactory loggerFactory)
+        {
+            builder
+                .RegisterInstance(loggerFactory)
+                .AsSelf()
+                .As<ILoggerFactory>();
+            builder
+                .RegisterGeneric(typeof(Logger<>))
+                .As(typeof(ILogger<>))
+                .SingleInstance();
+        }
+
+        public static void RegisterValidatorConfiguration(
+            this ContainerBuilder builder,
+            ValidatorConfiguration validatorConfig)
+        {
+            if (validatorConfig == null)
+            {
+                throw new ArgumentNullException(nameof(validatorConfig));
+            }
+
+            builder
+                .RegisterInstance(validatorConfig)
+                .AsSelf();
+        }
+
+        public static void RegisterEndpointConfiguration(
+            this ContainerBuilder builder,
+            EndpointConfiguration endpointConfig)
+        {
+            if (endpointConfig == null)
+            {
+                throw new ArgumentNullException(nameof(endpointConfig));
+            }
+
+            builder
+                .RegisterInstance(endpointConfig)
+                .AsSelf();
+        }
+
+        public static void RegisterMessageHandlerFactory(
+            this ContainerBuilder builder,
+            Func<HttpMessageHandler> messageHandlerFactory)
+        {
+            if (messageHandlerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(messageHandlerFactory));
+            }
+
+            builder
+                .RegisterInstance(messageHandlerFactory)
+                .AsSelf()
+                .As<Func<HttpMessageHandler>>();
+        }
```

Comment:
What't the difference here between `AsSelf()` and `As<Func<HttpMessageHandler>>()`?